pkg/controller/controller_instance.go

```diff
@@ -560,6 +593,12 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1beta1.ServiceIns
 		}
 	} else {
 		if instance.Status.CurrentOperation != v1beta1.ServiceInstanceOperationDeprovision {
+			if instance.Status.OrphanMitigationInProgress {
+				// There is no need in tracking orphan mitigation separately
+				// from the normal deletion
+				removeServiceInstanceCondition(instance, v1beta1.ServiceInstanceConditionOrphanMitigation)
+				instance.Status.OrphanMitigationInProgress = false
+			}
```

Comment:
I don't understand.

we're in the delete flow, with a timestamp, and not in deprovisioning state, 
the OM bool is set, ( ? so the source object is set from previous controller ? )

Reply:
@MHBauer this is the case when `instance.DeletionTimestamp != nil` i.e. the instance is marked for deletion.
Having `instance.Status.OrphanMitigationInProgress = true` means that before user deleted an instance, we have marked it for orphan mitigation but haven't finished yet.
Given that deletion has the highest priority over all other operations, keeping the orphan mitigation information is redundant at this point, so we clear it.